pkg/scheduler/scheduler.go

```diff
@@ -328,6 +328,10 @@ func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, e
 				klog.Errorf("Error preempting pod %v/%v: %v", victim.Namespace, victim.Name, err)
 				return "", err
 			}
+			// If the victim is a WaitingPod, send a reject message to the PermitPlugin
```

Comment:
It does not really make a difference technically, but it make more sense to first reject the pod and then delete it, this is semantically the right thing to do.

Reply:
Two pros for it from my perspective:

*  Let the permit plugin returns early after it receives `Reject` message, instead of waiting timeout.   

* When writing a [coscheduling](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/34-20180703-coscheduling.md) plugin which will permit pods until the number of pods reaches a certain number. Suppose that it requires 3 pods to run together, the permit plugin will return `Wait` for the first 2 pods, when it tries to permit the 3rd pod, it will find that there have been 2 pods in the waiting pod queue, then it will [`Allow`](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/v1alpha1/waiting_pods_map.go#L91) them all. It does not know whether or not they have been preempted. We need to add some check for it. With the PR, we could avoid these extra check.

Reply:
sorry, I guess my comment was not clear, I agree we should do this, I am just suggesting to do the rejection  before deleting the pod (i.e., move it up to line 327).

Reply:
:) It might fail to delete the pod(`sched.config.PodPreemptor.DeletePod()` might return an error), then we do not need to reject it, so I think we could reject after deleting successfully?

Reply:
right, it is just that doing it in this ordered will look strange though in the logs, we will see something like: 
- pod x deleted 
- pod x rejected

I don't have strong opinion on this though, so I am fine with this implementation.

Reply:
Yes, it will be a little strange if there are logs. There is no reject log, so it is not a problem now. :)